guides/source/engines.md

```diff
@@ -1090,43 +1369,82 @@ module MyApp
+`article_override.rb`) to clearly distinguish them from standard models and
+controllers and to avoid any conflicts with autoloaded files.
+
 #### Reopening Existing Classes Using `class_eval`
 
-For example, in order to override the engine model
+In order to override the engine model
 
 ```ruby
-# Blorgh/app/models/blorgh/article.rb
+# blorgh/app/models/blorgh/article.rb
 module Blorgh
   class Article < ApplicationRecord
     # ...
   end
 end
 ```
 
-you just create a file that _reopens_ that class:
+you can create a file that _reopens_ that class. This example reopens the
+`Blorgh::Article` model and adds a validation for the `title` attribute.
 
 ```ruby
-# MyApp/app/overrides/models/blorgh/article_override.rb
+# host_application/app/overrides/models/blorgh/article_override.rb
 Blorgh::Article.class_eval do
-  # ...
+  validates :title, presence: true, length: { minimum: 10 }
 end
 ```
 
-It is very important that the override _reopens_ the class or module. Using the `class` or `module` keywords would define them if they were not already in memory, which would be incorrect because the definition lives in the engine. Using `class_eval` as shown above ensures you are reopening.
+It is very important that the override _reopens_ the class or module. Using the
+`class` or `module` keywords would define them if they were not already in
+memory, which would be incorrect because the definition lives in the engine.
+Using
+[`class_eval`](https://api.rubyonrails.org/classes/Module.html#method-i-class_eval)
+as shown above ensures you are reopening an existing module.
 
 #### Reopening Existing Classes Using ActiveSupport::Concern
 
-Using `Class#class_eval` is great for simple adjustments, but for more complex
-class modifications, you might want to consider using [`ActiveSupport::Concern`]
-(https://api.rubyonrails.org/classes/ActiveSupport/Concern.html).
-ActiveSupport::Concern manages load order of interlinked dependent modules and
-classes at run time allowing you to significantly modularize your code.
+While
+[`Class#class_eval`](https://api.rubyonrails.org/classes/Module.html#method-i-class_eval)
+is useful for making simple runtime changes to a class, more complex
+modifications—especially those involving multiple modules with dependencies—are
+often better handled using
+[`ActiveSupport::Concern`](https://api.rubyonrails.org/classes/ActiveSupport/Concern.html).
+It provides a structured way to define module behavior and dependencies,
+ensuring consistent load order and making it easier to organize and compose
+reusable code.
 
-**Adding** `Article#time_since_created` and **Overriding** `Article#summary`:
+Suppose you want to extend the `Blorgh::Article` model from the host application
+by:
 
-```ruby
-# MyApp/app/models/blorgh/article.rb
+- Adding a new instance method: `time_since_created`
+- Overriding the existing `summary` method
 
+Here’s how you can do it cleanly using `ActiveSupport::Concern`:
+
+##### Adding/Overriding methods
```

Comment:
The order of the following sections is a bit confusing to me.
Instead of referencing an unknown concern that is created later, I think it would be clearer if we start with the concern.
After creating the concern we can include it in engine's Article.
We still need the engine's article as other users of the engine might not override it.
As a last step we can show the host application's Article.